src/plugins/intel_gpu/src/graph/graph_optimizer/propagate_constants.cpp

```diff
@@ -113,25 +136,51 @@ bool propagate_constants::has_non_const_user(program_node& node) const {
     return false;
 }
 
-std::list<std::pair<primitive_id, memory::ptr>> propagate_constants::calculate(engine& engine,
-                                                                               const ExecutionConfig& config,
-                                                                               std::shared_ptr<ov::threading::IStreamsExecutor> task_executor) {
+std::list<std::tuple<primitive_id, memory::ptr, std::shared_ptr<weightless_cache_manager>, std::shared_ptr<layout>>>
+propagate_constants::calculate(engine& engine,
+                               const ExecutionConfig& config,
+                               std::shared_ptr<ov::threading::IStreamsExecutor> task_executor) {
     if (!has_non_trivial_constants)
         return {};
 
     ExecutionConfig cf_config = config;
     cf_config.set_property(ov::intel_gpu::optimize_data(false));
     cf_config.set_property(ov::intel_gpu::custom_outputs(const_outputs));
     network::ptr net = network::build_network(engine, nodes, cf_config, task_executor, true);
-    for (auto& cin : const_inputs)
+    std::map<primitive_id, std::shared_ptr<weightless_cache_manager>> weightless_cache_map;
+    std::map<primitive_id, std::shared_ptr<layout>> weightless_cache_input_layout_map;
+    for (auto& cin : const_inputs) {
         net->set_input_data(cin->id(), cin->get_attached_memory_ptr());
 
+        auto users = cin->get_users();
+        if (users.size() == 1 && users.front()->is_type<reorder>()) {
+            auto rprim = users.front()->as<reorder>().get_primitive();
+            auto id = rprim->id;
+            auto cache_ptr = cin->as<data>().get_primitive()->cache_info;
+            weightless_cache_map.emplace(id, cache_ptr);
+            weightless_cache_input_layout_map.emplace(id, std::make_shared<layout>(cin->get_output_layout()));
+        }
+    }
+
     net->execute({});
     net->reset_execution(true);  // wait for computations to complete
     auto outputs = net->get_outputs();
 
-    std::list<std::pair<primitive_id, memory::ptr>> ret;
-    for (auto& out : outputs) ret.push_back({out->id(), out->output_memory_ptr()});
+    std::list<std::tuple<primitive_id, memory::ptr, std::shared_ptr<weightless_cache_manager>, std::shared_ptr<layout>>>
+        ret;
+    for (auto& out : outputs) {
+        std::shared_ptr<weightless_cache_manager> cache_ptr = nullptr;
+        std::shared_ptr<layout> in_layout = nullptr;
+        auto it = weightless_cache_map.find(out->id());
+        if (it != weightless_cache_map.end()) {
+            cache_ptr = it->second;
+        }
+        auto layout_it = weightless_cache_input_layout_map.find(out->id());
+        if (layout_it != weightless_cache_input_layout_map.end()) {
+            in_layout = layout_it->second;
+        }
```

Comment:
Can we use single map for both weightless_cache_manager and layout to avoid double lookup?

Reply:
Done